docs/6-rag/2-vector-stores.md

```diff
@@ -0,0 +1,171 @@
+# 📊 Vector Stores: The Foundation of RAG
+
+<div class="terminal-curl"></div>
+
+Think of vector stores as the specialized libraries that power RAG systems. While a traditional library organizes books alphabetically, vector stores organize information by **meaning**, allowing AI to find relevant content based on semantic similarity rather than just keyword matching.
+
+For educational platforms like Canopy, this means students can ask "How does bias affect hiring?" and get relevant information even if your course materials use terms like "discrimination in recruitment" or "algorithmic fairness in employment".
+
+## 🔍 What are Vector Stores?
+
+**Vector stores** (also called vector databases) are specialized databases that store and search through high-dimensional vectors - mathematical representations of text meaning. Here's how they work:
+
+
+![LLS RAG Architecture Diagram](images/rag5.png ':size=60%')
+
+### The Vector Magic ✨
+
+1. **Text → Numbers**: Every piece of text gets converted into a list of numbers (a vector) that captures its meaning
+2. **Similarity Search**: When you ask a question, the system finds vectors with similar "shapes" in the mathematical space
+3. **Lightning Fast**: Even with millions of documents, searches happen in milliseconds
+
+![LLS RAG Architecture Diagram](images/rag6.png)
+
+### Why Regular Databases Aren't Enough
+
+Let's use a simple example to understand the difference. Imagine you're building a pet care app with these documents:
+
+**❌ Traditional Keyword Search:**
+- Searching for "dog" only finds documents containing the exact word "dog"
+- Misses related documents about "puppy", "canine", "golden retriever", or "border collie"
+- A search for "cat care" won't find a document titled "Feline Health Tips"
+- Can't understand that "My furry friend loves to fetch" is about dogs
+
+**✅ Vector-Based Semantic Search:**
+- Searching for "dog" finds documents about puppies, specific breeds, and dog-related concepts
+- Understands that "feline" and "cat" are related, so "cat care" finds "Feline Health Tips"
+- Recognizes that "furry friend that fetches" is semantically similar to "dog behavior"
+- Ranks results by conceptual relevance: "golden retriever training" scores higher than "cat toys" for a "dog" search
+
+The magic happens because vectors capture *meaning*: "puppy", "canine", and "golden retriever" all get similar vector representations because they're conceptually related, even though they share no common letters!
+
```

Comment:
Just an idea, what do you think about having a hands-on example on how embeddings work before going into vector databases?